### PR TITLE
Chore: Break ControlShouldSupportTablePattern rule into two rules

### PR DESCRIPTION
**Pull request by RobGallo**
#### Describe the change

The old rule returned two possible `ErrorCode` values (in addition to `ErrorCode.Pass`), whereas each of the new separate rules only returns one. This makes it possible for us to specify the failure error code as part of the `RuleInfo` data, which will allow us to automate documentation for the rules regarding which error codes are returned by which rules.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/RulesTest/Library/ControlShouldSupportTablePatternInEdge.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            Assert.AreEqual(PropertyType.UIA_IsTablePatternAvailablePropertyId, this.Rule.Info.PropertyID);
+        }
+
+        [TestMethod]
+        public void ConditionMatchesExpectedElement()
+        {
+            int[] expectedTypes = { ControlType.Calendar, ControlType.Table };
+            var unexpectedTypes = ControlType.All.Difference(expectedTypes);
+            
+                var e = new MockA11yElement();
```
Comment (DaveTryon) [inline src/RulesTest/Library/ControlShouldSupportTablePatternInEdge.cs:54]: One extra indent on this line
```suggestion
            var e = new MockA11yElement();
```

File: src/RulesTest/Library/ControlShouldSupportTablePattern.cs [hunk 2]
```diff
@@ -52,5 +44,31 @@ public void CheckPropertyIdIsSet()
...
             Assert.AreEqual(PropertyType.UIA_IsTablePatternAvailablePropertyId, this.Rule.Info.PropertyID);
         }
+
+        [TestMethod]
+        public void ConditionMatchesExpectedElement()
+        {
+            int[] expectedTypes = { ControlType.Calendar, ControlType.Table };
+            var unexpectedTypes = ControlType.All.Difference(expectedTypes);
+            
+                var e = new MockA11yElement();
```
Comment (DaveTryon) [inline src/RulesTest/Library/ControlShouldSupportTablePattern.cs:54]: This line is indented one tab too far
```suggestion
            var e = new MockA11yElement();
```

File: src/Core/Enums/RuleId.cs [hunk 3]
```diff
@@ -126,6 +126,7 @@ public enum RuleId
         ControlShouldSupportSpreadsheetItemPattern,
         ControlShouldSupportTableItemPattern,
         ControlShouldSupportTablePattern,
+        ControlShouldSupportTablePatternInEdge,
```
Comment (DaveTryon) [inline src/Core/Enums/RuleId.cs:129]: @RobGallo, should we update the RuleDescriptions.md file as rules are modified or when we have a release? I tend to lean towards keeping the code and the description in sync in the repo, independent of the release schedule. Thoughts?

Comment (RobGallo) [reply src/Core/Enums/RuleId.cs:129]: I think we should keep them updated as code changes, not on release. I'll add that as a change.

File: src/Rules/Library/ControlShouldSupportTablePattern.cs [hunk 4]
```diff
@@ -25,15 +25,12 @@ public override EvaluationCode Evaluate(IA11yElement e)
         {
             if (e == null) throw new ArgumentNullException(nameof(e));
 
-            if (Patterns.Table.Matches(e))
-                return EvaluationCode.Pass;
-
-            return StringProperties.Framework.Is(Framework.Edge).Matches(e) ? EvaluationCode.Warning : EvaluationCode.Error;
+                return Patterns.Table.Matches(e) ? EvaluationCode.Pass : EvaluationCode.Error;
```
Comment (DaveTryon) [inline src/Rules/Library/ControlShouldSupportTablePattern.cs:28]: This row is indented with one extra tab
```suggestion
            return Patterns.Table.Matches(e) ? EvaluationCode.Pass : EvaluationCode.Error;
```

**Review by DaveTryon — APPROVED** [review]

**Pull request merged**